fdbserver/DataDistribution.actor.cpp

```diff
@@ -3543,13 +3567,18 @@ ACTOR Future<Void> storageServerTracker(
 				self->restartRecruiting.trigger();
 			}
 
+
 			if (lastIsUnhealthy && !status.isUnhealthy() &&
 			    ( server->teams.size() < targetTeamNumPerServer || self->lastBuildTeamsFailed)) {
 				self->doBuildTeams = true;
 				self->restartTeamBuilder.trigger(); // This does not trigger building teams if there exist healthy teams
 			}
 			lastIsUnhealthy = status.isUnhealthy();
 
+			if(server->ssVersionTooFarBehind.get() && !status.isUndesired) {
```

Comment:
When will this if condition be true?

There is no `wait`, except for the code path that marks the SS as failed, from the beginning of this actor to this line. 

Maybe I missed something?

Reply:
There is a wait statement nested in if statements on line 3556.
```wait(removeKeysFromFailedServer(cx, server->id, self->lock));```

Wouldn't this check be required if we hit that wait statement.

Reply:
Yeah, that's the wait I was referring to.

That wait won't be hit until the SS is marked as failed forcefully. This means the SS should have already got in the process of being removed. 

If you add an ASSERT(false) or SevError here, will it ever trigger?

Reply:
I will have to run simulation and then let you know.